devito/stencilkernel.py

```diff
@@ -0,0 +1,131 @@
+from hashlib import sha1
+from itertools import chain
+from os import path
+
+import cgen as c
+import numpy as np
+
+from devito.compiler import (get_compiler_from_env, get_tmp_dir,
+                             jit_compile_and_load)
+from devito.expression import Expression
+from devito.interfaces import SymbolicData
+from devito.iteration import Iteration
+from devito.logger import error
+from devito.tools import filter_ordered
+
+__all__ = ['StencilKernel']
+
+
+class StencilKernel(object):
+    """Code generation class, alternative to Propagator"""
```

Comment:
Based on this and the description of this PR, in the long term we expect the ``Propagator`` to be completely replaced by the ``StencilKernel`` right?